pkg/env/dev.go

```diff
@@ -84,30 +84,26 @@ func (d *dev) Listen() (net.Listener, error) {
 	return net.Listen("tcp", "localhost:8443")
 }
 
-func (d *dev) FPAuthorizer(tenantID, resource string) (refreshable.Authorizer, error) {
-	oauthConfig, err := adal.NewOAuthConfig(d.Environment().ActiveDirectoryEndpoint, tenantID)
-	if err != nil {
-		return nil, err
-	}
-
+func (d *dev) FPAuthorizer(tenantID string, scopes ...string) (autorest.Authorizer, error) {
 	fpPrivateKey, fpCertificates := d.fpCertificateRefresher.GetCertificates()
 
-	sp, err := adal.NewServicePrincipalTokenFromCertificate(*oauthConfig, d.fpClientID, fpCertificates[0], fpPrivateKey, resource)
+	options := d.Environment().ClientCertificateCredentialOptions()
+	tokenCredential, err := azidentity.NewClientCertificateCredential(
+		tenantID, d.fpClientID, fpCertificates, fpPrivateKey, options)
 	if err != nil {
 		return nil, err
 	}
 
-	return refreshable.NewAuthorizer(sp), nil
+	return azidext.NewTokenCredentialAdapter(tokenCredential, scopes), nil
 }
 
 func (d *dev) FPNewClientCertificateCredential(tenantID string) (*azidentity.ClientCertificateCredential, error) {
 	fpPrivateKey, fpCertificates := d.fpCertificateRefresher.GetCertificates()
 
-	credential, err := azidentity.NewClientCertificateCredential(tenantID, d.fpClientID, fpCertificates, fpPrivateKey, &azidentity.ClientCertificateCredentialOptions{
-		AuthorityHost:        d.Environment().AuthorityHost,
-		SendCertificateChain: true,
-	})
+	options := d.Environment().ClientCertificateCredentialOptions()
+	options.SendCertificateChain = true
```

Comment:
Do we also need AuthorityHost as an option here, or was that removed?

Reply:
**AuthorityHost** was removed in favor of a **Cloud** option in the embedded [ClientOptions](https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/azcore@v1.3.1/policy#ClientOptions) struct.

Essentially does the same thing: specifies which Azure cloud to use.  `d.Environment().ClientCertificateCredentialOptions()` sets it appropriately.